src/app/core/entity/model/entity.ts

```diff
@@ -191,8 +195,8 @@ export class Entity {
   /**
    * Get the class (Entity or the actual subclass of the instance) to call static methods on the correct class considering inheritance
    */
-  getConstructor(): EntityConstructor {
-    return <typeof Entity>this.constructor;
+  getConstructor<T extends Entity>(this: T): EntityConstructor<T> {
```

Comment:
what does this change mean?

Reply:
Prior to that calling `new Child().getConstructor()` would return a constructor of the type `EntityConstructor<Entity>`, so calling `new (new Child().getConstructor())()` would have the type `Entity` instead of `Child`. This change correctly returns the constructor for the subclass, in this case `EntityConstructor<Child>`.